Google button login

dahlgrenthomas opened the pull request:
This pull request introduces a google button that you can use to log in and see your cool name. Also stores your login data in the database so later we can set some of the attributes of that code to being "admin" or something.

dahlgrenthomas committed:
Add sign in and sign out buttons

You can now sign in with your google account but only if I've given you access to sign in since google likes security while in development. You can go into the console and see your account information and the id token that will need to eventually be sent to the server so it can identify what kind of user you are. The signout button signs you out but doesn't do anything else.

dahlgrenthomas committed:
Add better button integration to client

Instead of doing everything in the index html we now make a google button in the components html and its function now run in the ts file instead of all in the html so they can be tested.

dahlgrenthomas committed:
Add google branded button to website

Now the google button has the google G on it so it can be recognized by the user.

dahlgrenthomas committed:
Add http request to server for google sign in

Now the google button sends a request to the server containing an id token, the server will now need to have methods added so it can accept that token.

dahlgrenthomas committed:
Add files so a user can be added to the server database on login

Whenever you sign in a json will be put into the database with your information, the only problem is is that it will always add a new entry to the database instead of checking if you already exist.

dahlgrenthomas committed:
Add better google button

Now the button can be hovered with the mouse and give an effect, doesn't work on mobile view if screen is small enough it clips the text will need fixing.

dahlgrenthomas committed:
Add method to check if user is already in the database

Instead of adding a user to the database on every login it now checks if you are already in it.

dahlgrenthomas committed:
Add tests for google token checker

These tests add a google token that can be run to check the google token checker method and also will be used to test github secrets.

dahlgrenthomas committed:
Remove unused imports

dahlgrenthomas committed:
Remove logout() test

This test has been removed until a better way of testing it can be created by mocking a login.

dahlgrenthomas committed:
Fix the server tests for google token checker

The google token checker relied on a token you'd have to change every 60 minutes so I broke up the structure of the method so it could be tested without contacting google at all.

dahlgrenthomas committed:
Merge branch 'main' into GoogleButton

Getting ready for pull request

schu4276 reviewed (changes requested):
I think that we should fix the LGTM and better code hub errors before merging this.  Looks like we have to reduce the lines of code in the server, we can talk about how we want to do this because I do realize you only added a couple of lines of code.

halleypaulson reviewed (commented):
I think everything looks really good. It shouldn't be too much effort to fix the code hub so we should do that

dahlgrenthomas committed:
Fix lgtm alerts for merge

Removed all unused imports and verified the params in the comments of some functions that were old.

dahlgrenthomas committed:
Refactor code

Makes the code better for better code hub.
